api/controllers/linux/install/controller.go

```diff
@@ -256,17 +248,25 @@ func NewInstallController(opts ...InstallControllerOption) (*InstallController,
 		)
 	}
 
-	if controller.configValues != nil {
-		if controller.releaseData == nil || controller.releaseData.AppConfig == nil {
-			return nil, errors.New("app config not found")
+	if controller.releaseData != nil && controller.releaseData.AppConfig != nil {
+		// Initialize the app config manager if an app config is provided
+		if controller.appConfigManager == nil {
```

Comment:
shouldnt we always instantiate this and instead fail on nil controller.releaseData.AppConfig?